src/processors/file-handler.js

```diff
@@ -33,17 +33,22 @@ import {DATASET_FORMATS} from 'constants/default-settings';
 
 const BATCH_TYPE = {
   METADATA: 'metadata',
-  ROOT_OBJECT_COMPLETE: 'root-object-batch-complete'
+  PARTIAL_RESULT: 'partial-result',
+  FINAL_RESULT: 'final-result'
 };
 
 const CSV_LOADER_OPTIONS = {
-  batchSize: 4000, // Auto detect number of rows per batch (network batch size)
+  batchSize: 4000, // Auto de tect number of rows per batch (network batch size)
   rowFormat: 'object'
 };
 
 const JSON_LOADER_OPTIONS = {
-  _rootObjectBatches: true
-  // batchSize: 4000, // Auto detect number of rows per batch (network batch size)
+  // instruct loaders.gl on what json paths to stream
+  jsonpaths: [
+    '$', // JSON Row array
+    '$.features', // GeoJSON
+    '$.datasets' // KeplerGL JSON
```

Comment:
datasets is an array of datasets, which could be less than 10, we can make it be `$.datasets.0.data` so it always tries to stream the first dataset. And while we save in kepler, we can save the dataset with the longest rows first

Reply:
Agreed. The reason I left it as `$.datasets` for now is because it looks like the array indexing in jsonpaths isn't handled properly in loaders.gl.

Reply:
Yes, I think you can just add a test case in jsonpaths.spec and streaming-json-parser.spec in loaders.gl/json and it should be easy to pin down this little bug (at a guess, some string/number conversion issue perhaps, or openobject, openarray treated differently).